plugins/tedge_apt_plugin/src/main.rs

```diff
@@ -64,27 +74,60 @@ struct SoftwareModuleUpdate {
     pub path: Option<String>,
 }
 
+#[derive(Debug, Deserialize)]
+struct TedgeConfig {
+    pub apt: AptConfig,
+}
+
+#[derive(Debug, Deserialize)]
+struct AptConfig {
+    pub name: Option<String>,
+    pub maintainer: Option<String>,
+}
+
 fn run(operation: PluginOp) -> Result<ExitStatus, InternalError> {
     let status = match operation {
-        PluginOp::List {} => {
-            let apt = Command::new("apt")
-                .args(vec!["--manual-installed", "list"])
+        PluginOp::List { name, maintainer } => {
+            let dpkg_query = Command::new("dpkg-query")
+                .args(vec![
+                    "-f",
+                    "${Package}\t${Version}\t${Maintainer}\t${Status}\n",
+                    "-W",
+                ])
                 .stdout(Stdio::piped())
                 .spawn()
-                .map_err(|err| InternalError::exec_error("apt", err))?
+                .map_err(|err| InternalError::exec_error("dpkg-query", err))?
                 .wait_with_output()
-                .map_err(|err| InternalError::exec_error("apt", err))?;
+                .map_err(|err| InternalError::exec_error("dpkg-query", err))?;
+
+            let stdout = String::from_utf8(dpkg_query.stdout).unwrap_or_default();
+
+            let filter = match (&name, &maintainer) {
+                (None, None) => Regex::new(r"install ok installed"),
+
+                _ => Regex::new(
+                    format!(
+                        r"(^{}\t.*|^\S+\t\S+\t{}\s+.*)install ok installed",
+                        name.unwrap_or_default(),
+                        maintainer.unwrap_or_default()
+                    )
+                    .as_str(),
+                )
+                .map_err(|err| {
+                    eprintln!("{err}");
+                    std::process::exit(1)
```

Comment:
This error message is nicer now, but it would be perfect with some context.

```
target/debug/tedge-apt-plugin list --name '[' 
regex parse error:
    (^[\t.*|^\S+\t\S+\t\s+.*)install ok installed
      ^
error: unclosed character class
```

```suggestion
                    eprintln!("tedge-apt-plugin fails to list packages with matching name and maintener: {err}");
                    std::process::exit(1)
```